src/main/java/com/limechain/trie/structure/TrieStructure.java

```diff
@@ -500,13 +525,26 @@ public <U> boolean structurallyEquals(TrieStructure<U> other) {
         }
     }
 
-    public boolean deleteNodeAt(Nibbles key) {
-        return switch (existingNodeInner(key)) {
-            case ExistingNodeInnerResult.NotFound ignored -> false;
-            case ExistingNodeInnerResult.Found found -> deleteNodeAt(found.nodeIndex);
+    public boolean deleteStorageNodeAt(Nibbles key) {
+        return switch (node(key)){
+            case StorageNodeHandle<T> storageNodeHandle -> {
+                deleteStorageNodeAt(storageNodeHandle.rawNodeIndex);
+                yield true;
+            }
+            case BranchNodeHandle<T> branchNodeHandle -> false;
+            case Vacant<T> vacant -> false;
         };
     }
 
+    public boolean deleteInternalNodeAt(Nibbles key) {
+        Entry<T> entry = node(key);
+        if(!(entry instanceof Vacant<T>)){
+            deleteStorageNodeAt(entry.asNodeHandle().rawNodeIndex);
```

Comment:
This method we could rename to simply `deleteNodeAt` since it deletes the node with no checks.